app/src/main/java/com/lagradost/cloudstream3/syncproviders/providers/SimklApi.kt

```diff
@@ -0,0 +1,843 @@
+            debugPrint { "Full list update in ${this.name}." }
+            setKey(accountId, SIMKL_CACHED_LIST_TIME, lastRemoval)
+            getSyncListSince(null)
+        } else if (lastCacheUpdate < lastRealUpdate || lastCacheUpdate < lastScoreTime) {
+            debugPrint { "Partial list update in ${this.name}." }
+            setKey(accountId, SIMKL_CACHED_LIST_TIME, lastCacheUpdate)
+            AllItemsResponse.merge(getSyncListCached(), getSyncListSince(lastCacheUpdate))
+        } else {
+            debugPrint { "Cached list update in ${this.name}." }
+            getSyncListCached()
+        }
+        debugPrint { "List sizes: movies=${list?.movies?.size}, shows=${list?.shows?.size}, anime=${list?.anime?.size}" }
+
+        setKey(accountId, SIMKL_CACHED_LIST, list)
+
+        return list
+    }
+
+
+    override suspend fun getPersonalLibrary(): SyncAPI.LibraryMetadata? {
+        val list = getSyncListSmart() ?: return null
+
+        val baseMap =
+            SimklListStatusType.values()
+                .filter { it.value >= 0 && it.value != SimklListStatusType.ReWatching.value }
+                .associate {
+                    it.stringRes to emptyList<SyncAPI.LibraryItem>()
+                }
+
+        val syncMap = listOf(list.anime, list.movies, list.shows)
+            .flatten()
+            .groupBy {
+                it.status
+            }
+            .mapNotNull { (status, list) ->
+                val stringRes =
+                    status?.let { SimklListStatusType.fromString(it)?.stringRes } ?: return@mapNotNull null
+                val libraryList = list.map { it.toLibraryItem() }
+                stringRes to libraryList
+            }.toMap()
+
+        return SyncAPI.LibraryMetadata(
+            (baseMap + syncMap).map { SyncAPI.LibraryList(txt(it.key), it.value) }, setOf(
+                ListSorting.AlphabeticalA,
+                ListSorting.AlphabeticalZ,
+                ListSorting.UpdatedNew,
+                ListSorting.UpdatedOld,
+                ListSorting.RatingHigh,
+                ListSorting.RatingLow,
+            )
+        )
+    }
+
+    override fun getIdFromUrl(url: String): String {
+        val simklUrlRegex = Regex("""https://simkl\.com/[^/]*/(\d+).*""")
+        return simklUrlRegex.find(url)?.groupValues?.get(1) ?: ""
+    }
+
+    override suspend fun handleRedirect(url: String): Boolean {
+        val uri = url.toUri()
+        val state = uri.getQueryParameter("state")
+        // Ensure consistent state
+        if (state != lastLoginState) return false
+        lastLoginState = ""
+
+        val code = uri.getQueryParameter("code") ?: return false
+        val token = app.post(
+            "$mainUrl/oauth/token", json = TokenRequest(code)
+        ).parsedSafe<TokenResponse>() ?: return false
+
+        switchToNewAccount()
+        setKey(accountId, SIMKL_TOKEN_KEY, token.access_token)
+
+        val user = getUser()
```

Comment:
if getUser throws an error switchToOldAccount will not get called, will lead to invalid state, add normalsafeapicall